### DISCUSSION TITLE
Addition of a whereOrNull method

### DISCUSSION THREAD

**Discussion by sdsmith1981** [laravel/framework#45130] [Ideas]
I've found a lot of times in our code we were duplicating the same basic premise - a nested where that returns when a value matches or is null, so have added this to our service provider instead. 

Would this be of use, worth creating a PR for?

        Builder::macro('whereOrNull', function ($column, $value) {
            return $this->where(function ($query) use ($column, $value) {
                $query->where($column, $value);
                $query->orWhereNull($column);
            });
        });

**Comment by henzeb** [comment]
i'd suggest also adding an whereEmpty and a whereNotEmpty shortcut, as those are used often as well.

**Comment by ZebTheWizard** [comment]
`whereOrNull` and `orWhereNull` are awfully similar method names
would love to see this helper but with a slightly different name 
best I can come up with is `whereNullSafe` or `whereOptional`

**Reply by sdsmith1981** [reply]
I'm happy with `whereOptional`. I couldn't think of a better alternative tbh.
